src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.*;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GoPlaceholderCountInspection extends GoInspectionBase {
+  static final Pattern placeholderPattern =
+    Pattern.compile("(%((#)?(\\[\\d+\\]|\\*?)?\\.?)*((\\d*)(\\.\\d*))*\\s?(v|T|t|b|c|d|o|q|x|X|U|b|e|E|f|F|g|G|s|q|x|X|p))");
+
+  static final Pattern indexedPlaceholderPattern = Pattern.compile("(?:\\[(\\d+)\\])");
+
+  // This holds the name of the known formatting functions and position of the string to be formatted
+  private Map<String, Integer> formattingFunctions = new HashMap<String, Integer>();
+
+  public GoPlaceholderCountInspection() {
+    formattingFunctions.put("printf", 0);
+    formattingFunctions.put("sprintf", 0);
+    formattingFunctions.put("errorf", 0);
+    formattingFunctions.put("fprintf", 1);
+    formattingFunctions.put("fscanf", 1);
+    formattingFunctions.put("scanf", 0);
+    formattingFunctions.put("sscanf", 1);
+    formattingFunctions.put("fatalf", 0);
+    formattingFunctions.put("panicf", 0);
+  }
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitCallExpr(@NotNull GoCallExpr o) {
+        String name = o.getExpression().getText().toLowerCase();
```

Comment:
Please extract method `isFormattingFunction`

+

I think it's better to check real function declarations instead of call expressions.
To do that, resolve expression and then check whether it's a function with particular name from particular package. E.g.:

```java
func = callExpression.getExpression().getReference().resolve()
check func.getName
check func.getContainingFile.getPackage or func.getCointaningFile.getImportPath
```